src/Symfony/Component/HttpFoundation/Session/Storage/Handler/MongoDbSessionHandler.php

```diff
@@ -86,77 +94,92 @@ public function close(): bool
 
     protected function doDestroy(#[\SensitiveParameter] string $sessionId): bool
     {
-        $this->getCollection()->deleteOne([
-            $this->options['id_field'] => $sessionId,
-        ]);
+        $write = new BulkWrite();
+        $write->delete(
+            [$this->options['id_field'] => $sessionId],
+            ['limit' => 1]
+        );
+
+        $this->manager->executeBulkWrite($this->namespace, $write);
 
         return true;
     }
 
-    public function gc(int $maxlifetime): int|false
+    public function gc(int $maxlifetime): int
```

Comment:
We used getDeletedCount() before and had int|false as return type.

Now narrow it down to int. 

Was it wrong before?

Reply:
Indeed, I didn't realized that changing this return type would be breaking change. Reverted.